reqhandler/fields.go

```diff
@@ -0,0 +1,33 @@
+package reqhandler
+
+type FieldName string
+
+const (
+	ID       FieldName = "id"
+	QUERY              = "query"
```

Comment:
Because we use these constants as a key into the map of field lengths, and you can't use an integer as a key into a map, it has to be a string. Otherwise iota would be a good choice.